langstream-api/src/main/java/ai/langstream/api/runner/code/AgentContext.java

```diff
@@ -43,4 +43,9 @@ public interface AgentContext {
     default BadRecordHandler getBadRecordHandler() {
         return DEFAULT_BAD_RECORD_HANDLER;
     }
+
+    default void criticalFailure(Throwable error) {
+        System.err.printf("Critical failure: %s. Shutting down the runtime...", error.getMessage());
```

Comment:
for the default implementation we can simply log, this way the tests won't crash the JVM